src/Symfony/Component/Validator/Constraints/ChoiceValidator.php

```diff
@@ -59,6 +63,10 @@ public function validate(mixed $value, Constraint $constraint)
             $choices = $constraint->choices;
         }
 
+        if ($value instanceof \BackedEnum) {
```

Comment:
im not sure we should allow to mix&match different enum typed.

Reply:
We shouldn't. It's only related to BackedEnum. Needs to transform the BackedEnum value to a scalar type, this to work:
```php
final class ProductDto {
    #[Assert\Choice(choices: StatusEnum::class)]
    public StatusEnum $status;
}
```
Another enum type can't be passed and can't be validated with Assert\Choice, cause the exception will be [thrown](https://github.com/symfony/symfony/pull/44576/files#diff-d5a06a06b90acea359eabcd7589da0a955d0852c5b26b7417eb2a95aaa55ec65R39).
In case the value has UnitEnum type, validation won't be passed cause of value/choices type [mismatch](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Validator/Constraints/ChoiceValidator.php#L101).

Reply:
i mean, i can use any enum type and pass validation based on backed value. https://3v4l.org/dR1r7 

nevertheless, if the property a typed enum, you dont need a Choice constraint IMHO.

Reply:
Agree, there is no sense to validate the enum typed property with Assert\Choice. 
Have updated PR, check please.